MoreLinq/CountMethods.cs

```diff
@@ -0,0 +1,159 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2016 Leandro F. Vieira (leandromoh). All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Determines whether or not the number of elements in the sequence is greater than
+        /// or equal to the given integer.
+        /// </summary>
+        /// <typeparam name="T">Element type of sequence</typeparam>
+        /// <param name="source">The source sequence</param>
+        /// <param name="min">The minimum number of items a sequence must have for this
+        /// function to return true</param>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="min"/> is negative</exception>
+        /// <returns><c>true</c> if the number of elements in the sequence is greater than
+        /// or equal to the given integer or <c>false</c> otherwise.</returns>
+        /// <example>
+        /// <code>
+        /// var numbers = { 123, 456, 789 };
+        /// var result = numbers.AtLeast(2);
+        /// </code>
+        /// The <c>result</c> variable will contain <c>true</c>.
+        /// </example>
+        public static bool AtLeast<T>(this IEnumerable<T> source, int min)
+        {
+            if (min < 0) throw new ArgumentOutOfRangeException("min", "min must not be negative.");
```

Comment:
Exception messages must be full English sentences. Instead of:

    min must not be negative.

Consider instead:

    Count cannot be negative.

Please revise other exception message in the same light.